pyanaconda/modules/network/initialization.py

```diff
@@ -82,8 +82,13 @@ def run(self):
             log.debug("%s: No kickstart data.", self.name)
             return applied_devices
 
+        mainctx = create_new_context()
+        mainctx.push_thread_default()
```

Comment:
I think this should be handled with a context manager. We would need something like:

```
@contextmanager
def nm_client_in_thread():
    mainctx = create_new_context()
    mainctx.push_thread_default()

    try:
        yield get_new_nm_client()
    finally:
        mainctx.pop_thread_default()

```
Then the tasks could use the following template:

```
def run(self):
   ...
   with nm_client_in_thread() as nm_client:
      self._nm_client = nm_client
      return self._run()

def _run(self):
   ...

```

Reply:
I will fix this soon.

Reply:
Addressed in https://github.com/rhinstaller/anaconda/pull/4322
Well, I am not using the _run method, if you think it would be better you can comment in the new PR.